### DISCUSSION TITLE
Is there an example how to create a VSIX package for a langium language?

### DISCUSSION THREAD

**Discussion by goto40** [eclipse-langium/langium#650] [Q&A] [answered]
Thank you very much for langium 0.4.0. It really runs smoothly and the documentation is very useful!

Is there an example somewhere how to produce a VSIX package?

**Comment by msujew** [comment, marked as answer]
Hey @goto40,

thanks for the compliment, it's always great to hear positive feedback from our users :)

> Is there an example somewhere how to produce a VSIX package?

Actually yes. Since [Langium is written using Langium](https://github.com/langium/langium/blob/main/packages/langium/src/grammar/langium-grammar.langium), one such example would be our [langium-vscode](https://github.com/langium/langium/tree/main/packages/langium-vscode) extension. Note that you should bundle your extension and language server code before producing a VSIX package. See our [code bundling guide](https://langium.org/guides/code-bundling/) for more info. Aside from bundling your code, all you have to do is run `vsce package` in your extension directory.

**Reply by goto40** [reply]
Thank you for this link. Together with the langium documentation I have a good starting point.

**Comment by luan-xiaokun** [comment]
Just as a supplement, I also found this [vscode's guide](https://code.visualstudio.com/api/working-with-extensions/bundling-extension) useful, especially if you want to reduce your VSIX package size.
